src/commands/up/index.ts

```diff
@@ -72,107 +78,126 @@ async function pullDockerFiles (tag: string) {
-    default:
-      throw new Error(`ops does not currently support ${arch}`);
-  }
+function runFrontend (dependencies: string, open: any) {
+  logger.info('Launching frontend on 0.0.0.0:3000...');
+  const commands = [
+    `docker build --pull --build-arg RUNTIME_DEPENDENCIES=${dependencies} -f ${UI_FILEPATH} -t ops-frontend .`,
+    `docker container stop ops-frontend || ${EXIT_0}`,
+    `docker container rm ops-frontend || ${EXIT_0}`,
+    'docker run --name ops-frontend --env AWS_REGION=us-west-2 --env API_ENDPOINT=http://ops-api:8000 -i -p 3000:3000 --network=ops-console ops-frontend;'
+  ].join(';\n');
+  const childProcess = runCommand(commands);
+  childProcess.stdout.on('data', (data) => {
+    if (data.includes(FRONTEND_SUCCESS_INDICATOR )) {
+      logger.success('Ops console frontend is now running on http://0.0.0.0:3000');
+      open.default('http://0.0.0.0:3000');
+    }
+  });
+  return childProcess;
 }
 
-function validateConfigFilePath (configFile: string) {
-  const absolutePath = path.resolve(configFile || `${process.cwd()}/${DEFAULT_CONFIG_FILENAME}`);
-  if (fs.existsSync(absolutePath)) {
-    return {
-      dir: path.dirname(absolutePath),
-      file: path.basename(absolutePath)
-    };
-  }
-  throw new Error(`Specified config file ${absolutePath} does not exist.`);
+function cleanup () {
+  logger.info('Cleaning up...');
+  fs.unlink(API_FILEPATH, () => { return; });
+  fs.unlink(UI_FILEPATH, () => { return; });
+}
+
+const signals = [ 'SIGINT', 'SIGQUIT', 'SIGTERM' ];
+function setParentCleanupHandler () {
+  signals.forEach((signal) => {
+    process.on(signal, cleanup);
+  });
 }
 
-function handleExitSignalCleanup (backendProcess?: ChildProcess, frontendProcess?: ChildProcess) {
-  function cleanup () {
-    logger.info('Cleaning up...');
-    fs.unlink(API_FILEPATH, () => { return; });
-    fs.unlink(UI_FILEPATH, () => { return; });
+function setProcessCleanupHandler (backendProcess: ChildProcess, frontendProcess: ChildProcess) {
+  function cleanupProcesses () {
+    runCommand(`docker stop ops-frontend ops-api || ${EXIT_0}; docker network rm ops-console || ${EXIT_0}`);
     backendProcess?.kill();
     frontendProcess?.kill();
-    runCommand(`docker stop ops-frontend ops-api || ${EXIT_0}; docker network rm ops-console || ${EXIT_0}`);
   }
-  process.on('SIGINT', cleanup);
-  process.on('SIGQUIT', cleanup);
-  process.on('SIGTERM', cleanup); 
+
+  signals.forEach((signal) => {
+    process.on(signal, cleanupProcesses);
+  });
+  backendProcess?.on('error', () => {
+    logger.error('Failed to launch ops console backend!');
+    cleanup();
+    cleanupProcesses();
+  });
+  frontendProcess?.on('error', () => {
+    logger.error('Failed to launch ops console frontend!');
+    cleanup();
+    cleanupProcesses();
+  });
+  backendProcess?.on('exit', (code) => {
+    if (!isNil(code) && code !== 0) {
+      logger.error(`Backend exited with code ${code}`);
+      cleanup();
+      cleanupProcesses();
+    }
+  });
```

Comment:
this makes sense, I guess the issue comes in with how we want to error handle exec failures. I think it can be hard to tell why the command failed except by parsing stdout/stderr from exec. With the flow added to this pr, without running the verbose flag, a user will be informed of the exit code and be asked to rerun using the -V flag to debug further. While rerunning, the docker logs would likely tell the user that the port is in use. We could catch the error and display a more accurate error log but I don't think looking at the exit code would be enough since a 125 seems generic (could be permissions issue). The alternative would be parsing stderr for a string saying the port is in use. We could add this parsed error handling as we come across more errors (although I'm a little worried about the overhead of checking each log for different matches). Don't mind making this addition if we think it's valuable, but I think asking the user to run with -V is decent. Maybe we could do a precheck for errors like this that we expect, kind of like the dependency validation?